engine/src/main/java/org/terasology/engine/rendering/nui/internal/LwjglCanvasRenderer.java

```diff
@@ -100,6 +88,9 @@
     private DisplayDevice displayDevice;
     private float uiScale = 1f;
 
+    private Matrix4fStack modelMatrixStack = new Matrix4fStack(100000);
```

Comment:
umm used an arbitrarily large number for the max stack size.

Reply:
This will actually allocate 100000 matrices. That's a few MB of space (and also, having lots of small objects may worsen garbage-collection performance even if they are all long-lived, although I don't know whether that still true if each of them is pretty self-contained like this).